File: JavaScript solutions/459. Repeated Substring Pattern.js
/**
 * @param {string} s
 * @return {boolean}
 */
var repeatedSubstringPattern = function (s) {
  const size = s.length;
  const postfix = s.substring(1);
  const prefix = s.substring(0, size - 1);

  const newS = postfix + prefix;
  return newS.includes(s);
};

File: C++ solutions/459. Repeated Substring Pattern.cpp
class Solution {
public:
    bool repeatedSubstringPattern(string s) {
        int len= s.size();
        string postfix = s.substr(1, len-1);
        string prefix = s.substr(0, len-1);
        
        string newS= postfix + prefix;
        return newS.find(s) != string::npos;
    }
};